### DISCUSSION TITLE
What do base layer bottom pinkie and ring fingers do when held?

### DISCUSSION THREAD

**Discussion by Lenbok** [manna-harbour/miryoku#105] [Q&A] [answered]
I'm using Miryoku ZMK on a corne, with qwerty as my default. On the bottom row most of the keys repeat if held, except z, x, ., and /. They have some kind of hold behaviour but I can't see it described anywhere. Is it possible to turn these off so that they act like regular keys?

**Comment by manna-harbour** [comment, marked as answer]
Pinkie is the Button layer, see https://github.com/manna-harbour/miryoku/tree/master/docs/reference#layers and https://github.com/manna-harbour/miryoku/tree/master/docs/reference#button. It's for use with pointing devices, particularly buttonless or integrated, but can also be used for one handed clipboard.

Ring is AltGr. That's needed for international layouts such as EurKEY or US International. It's not shown on the diagrams at the moment as I'm planning on making it optional, see https://github.com/manna-harbour/miryoku/issues/33. Later it'll be added to the docs when they're expanded. And also will be included in the diagrams when they are auto generated, see https://github.com/manna-harbour/miryoku/issues/72. It can be seen now in the source tables, see https://github.com/manna-harbour/miryoku_babel#hold, or in the generated source, e.g. https://github.com/manna-harbour/miryoku_zmk/blob/aa6bb9f674f13a3a8bb04de99b5f0090df00cac9/miryoku/miryoku_babel/miryoku_layer_alternatives.h#L120.

To disable them you can substitute a custom Base layer, see https://github.com/manna-harbour/miryoku/discussions/85.

In general the Base layer keys don't auto repeat, as many are tap hold. The thumbs have basic versions on the layers that can be used for auto repeat, such as backspace and delete on Nav (although for those, combining with control is generally preferable). For the alphas, the cursor keys on Nav can be used instead of holding for navigation in vi, and the Tap layer can be used for games etc.

**Reply by Valodim** [reply]
thanks for doing such great maintenance for this layout, it's quite inspiring!

I'd like to report that the altgr key not being documented tripped me up quite a bit, and I imagine it's similar for other users of us intl or similar layouts who need their language's special characters. It would be really helpful to mention it in the reference, even just as a placeholder.
